src/Microsoft.SqlTools.ServiceLayer/Connection/CancelTokenKey.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.ServiceLayer.Connection.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.Connection
+{
+    /// <summary>
+    /// Used to uniquely identify a CancellationTokenSource associated with both
+    /// a string URI and a string connection type.
+    /// </summary>
+    public class CancelTokenKey : CancelConnectParams
+    {
+        public override bool Equals(object obj)
+        {
+            if (!(obj is CancelTokenKey))
+            {
+                return false;
+            }
+
+            CancelTokenKey other = obj as CancelTokenKey;
+            return String.Equals(other.OwnerUri, OwnerUri) && String.Equals(other.Type, Type);
```

Comment:
>String [](start = 19, length = 6)

we should use "string" instead of "String" when access methods in this code...i.e., prefer calling methods against the primitive type instead of the corresponding BCL class.